types/stylelint/index.d.ts

```diff
@@ -259,7 +259,7 @@ declare namespace stylelint {
 	};
 
 	type FixerData = {
-		range: Range;
+		range: Range | void;
```

Comment:
`void` and `undefined` are not equivalent.
The current type warns if you return anything but `undefined` as illustrated by the `keyframe-selector-notation`.
i.e. it would require to explicitly `return undefined` in fixers, which is not desirable.
It's not ideal to have the `void` type of `fix` flow to `range` but that's convenient.

Reply:
Still, I feel it weird to use `void` for non-function return types. To resolve type errors by the `Range | undefined` change, we can just set `range: undefined` explicitly in `addFixData()`:

```diff
-	array.push({ range, fixed });
+	array.push(range ? { range, fixed } : { range: undefined, fixed});
```

Reply:
[f45d6ab](https://github.com/stylelint/stylelint/pull/7748/commits/f45d6abf9f1109aac2fdb8112e9ee1d95c17b34b)